research/train.py

```diff
@@ -46,22 +55,25 @@ def _update(engine, batch):
         optimizer.step()
 
         # Anything you want to log must be returned in this dictionary
-        update_dict = {"nll": loss.item(), "y_pred": y_pred, "y": y}
+        update_dict = {
+            "nll": loss.item(),
+            "y_pred": y_pred,
+            "y": y,
+            "im": (inverse_mnist_preprocess(x)[0] * 255).type(torch.uint8).squeeze()
+        }
```

Comment:
Same here we might want the config to specify the output for the logging in the config

Reply:
See my comments above, I don't agree. You may pick up a bit of _elegance_ in your code, but you lose a lot of clarity, because now you have to flip back and forth between your training file and your config. The thing I really want to avoid is creating a bunch of mini-API's inside the code because they drive people insane. A good example is regex - most people have to look up the docs to use it every time. My hope is our naming, and usage is clear enough that you don't have to look through the whole codebase to understand how to log.